src/ray/common/ray_config.cc

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ray/common/ray_config.h"
+
+#include <sstream>
+#include <typeinfo>
+
+#include "nlohmann/json.hpp"
+#include "ray/util/logging.h"
+
+using json = nlohmann::json;
+
+RayConfig &RayConfig::instance() {
+  static RayConfig config;
+  return config;
+}
+
+/// -----------Include ray_config_def.h to set config items.-------------------
+/// A helper macro that helps to set a value to a config item.
+#define RAY_CONFIG(type, name, default_value) \
+  if (pair.key() == #name) {                  \
+    name##_ = pair.value().get<type>();       \
+    continue;                                 \
+  }
```

Comment:
Does the unuesd default_value can be removed? 
Is it possible to replace the RAY_CONFIG with a function?

Reply:
No, it cannot be removed. The `RAY_CONFIG` is a macro and it will be expanded several times in `ray_config.h` and `ray_config.cc` with different templates, so it cannot be replaced with a function.